docs/conf.py

```diff
@@ -9,9 +9,42 @@
 # -- Project information -----------------------------------------------------
 import sys
 import os
-sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
+
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))
 import modin
 
+
+def hide_lines(path, patterns_to_hide, is_unhide=False):
+    walker = os.walk(path)
+
+    for root, _, files in walker:
+        for file in files:
+            if not file.endswith(".py"):
+                continue
+
+            file_name = os.path.join(root, file)
+            with open(file_name, "r") as fd:
+                file_text = fd.readlines()
+                for i, line in enumerate(file_text):
+                    for pattern in patterns_to_hide:
+                        if is_unhide:
+                            if pattern in line:
+                                file_text[i] = line[1:]
+                        else:
+                            if pattern in line:
+                                file_text[i] = "#" + line
+            with open(file_name, "w") as fd:
+                fd.writelines(file_text)
+
+
+hide_lines(
+    os.path.abspath(
+        os.path.join(os.path.dirname(__file__), "../modin/engines/ray/cudf_on_ray")
+    ),
+    ["import cudf", "import cupy", "instance_type = cudf.DataFrame"],
+    is_unhide=False,
+)
+
```

Comment:
@YarShev @vnlitvinov please, take a look.

To build docstrings of `cuDF` engine files using autodoc we need to install `cudf` and `cupy` conda packages from nVidia channels ([link](https://github.com/rapidsai/cudf#conda)). I excluded this way for now by the next reasons:

1. The size of all installed packages is about 2GB.
2. nVidia GPU driver is require to install (possible, issues will be occured on the server without GPU).
3. If install `cudf` only without dependencies, it stills required nvidia GPU driver binaries.

I added temporary draft solution to hide lines in the code with cudf-specific modules/keywords. This solution is added to `docs/conf.py` and run before `autodoc` builds docstrings. Built docs is correct in this case. But user code is changed after building documentation. Is it there any ways to do operation after building docs automatically?

What do you think about it? Have any ideas how to resolve this?

Reply:
We should be able to create dummpy `cudf` and `cupy` modules in `conf.py` if they're not installed, let me handle that.

Reply:
Should be fixed by 6002806